### PR TITLE
feat: add intrange linter

### PR DESCRIPTION
**Pull request by ckaznocha**
Hi,

I'd like to add a new linter: [intrange](https://github.com/ckaznocha/intrange)

It looks for loops which iterate over integers and suggests that the new go1.22 support for ranging over integers can be used. See: https://go.dev/ref/spec#Go_1.22

It **does not** suggest updating loops which:
- increment more than 1 in each iteration
- mutate the value of the integer in the body of the loop
- use anything other than an integer type
- start with anything other than zero

It works for integer literals, const/vars, and expressions that evaluate to an integer

### CODE REVIEW

**Comment by ldez** [comment]
/
In order for a pull request adding a linter to be reviewed, the linter and the PR must follow some requirements.

- [ ] The CLA must be signed

### Pull Request Description

- [ ] It must have a link to the linter repository.
- [ ] It must provide a short description of the linter.

### Linter

- [ ] It must not be a duplicate of another linter or a rule of a linter. (the team will help to verify that)
- [ ] It must have a valid license (AGPL is not allowed) and the file must contain the required information by the license, ex: author, year, etc.
- [ ] The linter repository must have a CI and tests.
- [ ] It must use [`go/analysis`](https://golangci-lint.run/contributing/new-linters/).
- [ ] It must have a valid tag, ex: `v1.0.0`, `v0.1.0`.
- [ ] It must not contain `init()`.
- [ ] It must not contain `panic()`.
- [ ] It must not contain `log.fatal()`, `os.exit()`, or similar.
- [ ] It must not modify the AST.
- [ ] It must not have false positives/negatives. (the team will help to verify that)
- [ ] It must have tests inside golangci-lint.

### The Linter Tests Inside Golangci-lint

- [ ] They must have at least one std lib import.
- [ ] They must work with `T=<name of the linter test file>.go make test_linters`. (the team will help to verify that)

### `.golangci.reference.yml`

- [ ] The linter must be added to the list of available linters (alphabetical case-insensitive order).
    - `enable` and `disable` options
- [ ] If the linter has a configuration, the exhaustive configuration of the linter must be added (alphabetical case-insensitive order)
    - The values must be different from the default ones.
    - The default values must be defined in a comment.
    - The option must have a short description.

### Others Requirements

- [ ] The files (tests and linter) inside golangci-lint must have the same name as the linter.
- [ ] The `.golangci.yml` of golangci-lint itself must not be edited and the linter must not be added to this file.
- [ ] The linters must be sorted in the alphabetical order (case-insensitive) in the `Manager.GetAllSupportedLinterConfigs(...)` and `.golangci.reference.yml`.
- [ ] The load mode (`WithLoadMode(...)`):
    - if the linter doesn't use types: `goanalysis.LoadModeSyntax`
    - `goanalysis.LoadModeTypesInfo` required `WithLoadForGoAnalysis()` in the `Manager.GetAllSupportedLinterConfigs(...)`
- [ ] The version in `WithSince(...)` must be the next minor version (`v1.X.0`) of golangci-lint.
- [ ] `WithURL()` must contain the URL of the repository.

### Recommendations

- [ ] The linter should not use SSA. (SSA can be a problem with generics)
- [ ] The linter repository should have a readme and linting.
- [ ] The linter should be published as a binary. (useful to diagnose bug origins)

---

The golangci-lint team will edit this comment to check the boxes before and during the review.

The code review will start after the completion of those checkboxes (except for the specific items that the team will help to verify).

If the author of the PR is a member of the golangci-lint team, he should not edit this message.

**This checklist does not imply that we will accept the linter.**

**Comment by ldez** [comment]
FYI 0.0.x are not semver complaint because a bug fix cannot happen before a minor release or an impossible initial version v0.0.0.

> PATCH version when you make backward compatible bug fixes

https://semver.org/

> How should I deal with revisions in the 0.y.z initial development phase?
>
> The simplest thing to do is start your initial development release at 0.1.0 and then increment the minor version for each subsequent release.

https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase

---

The version in `WithSince(...)` must be the next minor version (v1.X.0) of `golangci-lint`.

**Comment by ckaznocha** [comment]
Thanks for taking a first pass @ldez! I believe both of your comments have been addressed.

File: pkg/golinters/intrange.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+func NewIntrange() *goanalysis.Linter {
+	a := intrange.Analyzer
+
+	return goanalysis.NewLinter(
+		a.Name,
+		a.Doc,
+		[]*analysis.Analyzer{a},
+		nil,
+	).WithLoadMode(goanalysis.LoadModeTypesInfo)
```
Comment (ldez) [inline pkg/golinters/intrange.go:18]:
```suggestion
	).WithLoadMode(goanalysis.LoadModeSyntax)
```

File: pkg/lint/lintersdb/manager.go [hunk 2]
```diff
@@ -608,6 +608,11 @@ func (m Manager) GetAllSupportedLinterConfigs() []*linter.Config {
 			WithURL("https://github.com/mvdan/interfacer").
 			Deprecated("The repository of the linter has been archived by the owner.", "v1.38.0", ""),
 
+		linter.NewConfig(golinters.NewIntrange()).
+			WithSince("v1.57.0").
+			WithLoadForGoAnalysis().
```
Comment (ldez) [inline pkg/lint/lintersdb/manager.go:613]:
```suggestion
```

**Review by ldez — CHANGES_REQUESTED** [review]

**Review by ldez — APPROVED** [review]
LGTM

**Pull request merged**